patches/api/0483-Change-Art-to-an-interface.patch

```diff
@@ -0,0 +1,287 @@
++    Art BACKYARD = getArt("backyard");
++    Art BOUQUET = getArt("bouquet");
++    Art CAVEBIRD = getArt("cavebird");
++    Art CHANGING = getArt("changing");
++    Art COTAN = getArt("cotan");
++    Art ENDBOSS = getArt("endboss");
++    Art FERN = getArt("fern");
++    Art FINDING = getArt("finding");
++    Art LOWMIST = getArt("lowmist");
++    Art ORB = getArt("orb");
++    Art OWLEMONS = getArt("owlemons");
++    Art PASSAGE = getArt("passage");
++    Art POND = getArt("pond");
++    Art SUNFLOWERS = getArt("sunflowers");
++    Art TIDES = getArt("tides");
+ 
+-    private final int id, width, height;
+-    private final NamespacedKey key;
+-    private static final HashMap<String, Art> BY_NAME = Maps.newHashMap();
+-    private static final HashMap<Integer, Art> BY_ID = Maps.newHashMap();
++    private static Art getArt(final String id) {
++        return Registry.ART.get(NamespacedKey.minecraft(id));
++    }
+ 
+-    private Art(int id, int width, int height) {
+-        this.id = id;
+-        this.width = width;
+-        this.height = height;
+-        this.key = NamespacedKey.minecraft(name().toLowerCase(Locale.ROOT));
++    /**
++     * @deprecated use {@link Registry#ART}
++     */
++    @Deprecated(since = "1.21.1", forRemoval = true)
++    static @NotNull Art valueOf(final @NotNull String name) {
++        final Art art = Registry.ART.get(NamespacedKey.fromString(name.toLowerCase(Locale.ROOT)));
++        Preconditions.checkArgument(art != null, "No art found with the name %s", name);
++        return art;
++    }
++
++    /**
++     * @deprecated Use {@link Registry#ART}
++     */
++    @Deprecated(since = "1.21.1", forRemoval = true)
++    static Art[] values() {
++        return com.google.common.collect.Lists.newArrayList(Registry.ART).toArray(Art[]::new);
+     }
++    // Paper end - make interface
+ 
+     /**
+      * Gets the width of the painting, in blocks
+      *
+      * @return The width of the painting, in blocks
+      */
+-    public int getBlockWidth() {
+-        return width;
+-    }
++    public int getBlockWidth(); // Paper - make interface
+ 
+     /**
+      * Gets the height of the painting, in blocks
+      *
+      * @return The height of the painting, in blocks
+      */
+-    public int getBlockHeight() {
+-        return height;
+-    }
++    public int getBlockHeight(); // Paper - make interface
+ 
+     /**
+      * Get the ID of this painting.
+      *
+      * @return The ID of this painting
+-     * @apiNote Internal Use Only
++     * @deprecated use {@link #getKey()}
+      */
+-    @org.jetbrains.annotations.ApiStatus.Internal // Paper
+-    public int getId() {
+-        return id;
+-    }
++    @Deprecated(forRemoval = true, since = "1.21.1") // Paper - deprecate
++    public int getId(); // Paper - make interface
+ 
+     @NotNull
+     @Override
+-    public NamespacedKey getKey() {
+-        return key;
+-    }
++    public NamespacedKey getKey(); // Paper - make interface
```

Comment:
Should this be deprecated here or in the other patch for inlined painting variant?